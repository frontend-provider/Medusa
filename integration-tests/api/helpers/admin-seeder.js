const Scrypt = require("scrypt-kdf")
const { User } = require("@medusajs/medusa")

module.exports = async (dataSource, data = {}) => {
  const manager = dataSource.manager

  const buf = await Scrypt.kdf("secret_password", { logN: 1, r: 1, p: 1 })
  const password_hash = buf.toString("base64")

  await manager.insert(User, {
    id: "admin_user",
    email: "admin@medusa.js",
    api_token: "test_token",
    role: "admin",
    password_hash,
    ...data,
  })
}
