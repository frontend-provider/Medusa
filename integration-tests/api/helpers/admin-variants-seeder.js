const {
  Region,
  Product,
  ProductVariant,
  PriceList,
  CustomerGroup,
  Customer,
  Image,
  ShippingProfile,
  ProductCollection,
  ProductOption,
} = require("@medusajs/medusa")

module.exports = async (dataSource, data = {}) => {
  const manager = dataSource.manager

  const yesterday = ((today) => new Date(today.setDate(today.getDate() - 1)))(
    new Date()
  )

  const tenDaysAgo = ((today) => new Date(today.setDate(today.getDate() - 10)))(
    new Date()
  )

  const defaultProfile = await manager.findOne(ShippingProfile, {
    where: { type: ShippingProfile.default },
  })

  const collection = manager.create(ProductCollection, {
    id: "test-collection",
    handle: "test-collection",
    title: "Test collection",
  })

  await manager.save(collection)

  await manager.insert(Region, {
    id: "reg-europe",
    name: "Test Region Europe",
    currency_code: "eur",
    tax_rate: 0,
  })

  await manager.insert(Region, {
    id: "reg-us",
    name: "Test Region US",
    currency_code: "usd",
    tax_rate: 0,
  })

  const customer = await manager.create(Customer, {
    id: "test-customer",
    email: "john@doe.com",
    first_name: "John",
    last_name: "Doe",
    password_hash:
      "c2NyeXB0AAEAAAABAAAAAVMdaddoGjwU1TafDLLlBKnOTQga7P2dbrfgf3fB+rCD/cJOMuGzAvRdKutbYkVpuJWTU39P7OpuWNkUVoEETOVLMJafbI8qs8Qx/7jMQXkN", // password matching "test"
    has_account: true,
  })

  const customerGroup = await manager.create(CustomerGroup, {
    id: "test-group",
    name: "test-group",
  })

  await manager.save(customerGroup)
  customer.groups = [customerGroup]
  await manager.save(customer)

  const priceListActive = await manager.create(PriceList, {
    id: "pl",
    name: "VIP sale",
    description: "All year sale for VIP customers.",
    type: "sale",
    status: "active",
  })

  await manager.save(priceListActive)

  const priceListExpired = await manager.create(PriceList, {
    id: "pl_expired",
    name: "VIP summer sale",
    description: "Summer sale for VIP customers.",
    type: "sale",
    status: "active",
    starts_at: tenDaysAgo,
    ends_at: yesterday,
  })

  await manager.save(priceListExpired)

  const priceListWithCustomers = await manager.create(PriceList, {
    id: "pl_with_customers",
    name: "VIP winter sale",
    description: "Winter sale for VIP customers.",
    type: "sale",
    status: "active",
  })

  priceListWithCustomers.customer_groups = [customerGroup]

  await manager.save(priceListWithCustomers)

  const productMultiReg = manager.create(Product, {
    id: "test-product",
    handle: "test-product-reg",
    title: "Multi Reg Test product",
    profile_id: defaultProfile.id,
    description: "test-product-description",
    status: "published",
    collection_id: "test-collection",
  })

  const image = manager.create(Image, {
    id: "test-image",
    url: "test-image.png",
  })

  productMultiReg.images = [image]

  await manager.save(productMultiReg)

  await manager.save(ProductOption, {
    id: "test-option",
    title: "test-option",
    product_id: "test-product",
  })

  const variantMultiReg = await manager.create(ProductVariant, {
    id: "test-variant",
    inventory_quantity: 10,
    title: "Test variant",
    variant_rank: 0,
    sku: "test-sku",
    ean: "test-ean",
    upc: "test-upc",
    barcode: "test-barcode",
    product_id: "test-product",
    prices: [
      {
        id: "test-price-multi-usd",
        currency_code: "usd",
        type: "default",
        amount: 100,
      },
      {
        id: "test-price-discount-multi-usd",
        currency_code: "usd",
        amount: 80,
        price_list_id: "pl",
      },
      {
        id: "test-price-discount-expired-multi-usd",
        currency_code: "usd",
        amount: 70,
        price_list_id: "pl_expired",
      },
      {
        id: "test-price-multi-eur",
        currency_code: "eur",
        amount: 100,
      },
      {
        id: "test-price-discount-multi-eur",
        currency_code: "eur",
        amount: 80,
        price_list_id: "pl",
      },
      {
        id: "test-price-discount-multi-eur-with-customer",
        currency_code: "eur",
        amount: 40,
        price_list_id: "pl_with_customers",
      },
      {
        id: "test-price-discount-expired-multi-eur",
        currency_code: "eur",
        amount: 70,
        price_list_id: "pl_expired",
      },
    ],
    options: [
      {
        id: "test-variant-option-reg",
        value: "Default variant",
        option_id: "test-option",
      },
    ],
  })

  await manager.save(variantMultiReg)
}
