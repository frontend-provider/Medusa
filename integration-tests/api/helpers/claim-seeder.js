const {
  ClaimOrder,
  Order,
  LineItem,
  Fulfillment,
  Return,
} = require("@medusajs/medusa")

module.exports = async (dataSource, data = {}) => {
  const manager = dataSource.manager

  let orderWithClaim = manager.create(Order, {
    id: "order-with-claim",
    customer_id: "test-customer",
    email: "test@email.com",
    payment_status: "captured",
    fulfillment_status: "fulfilled",
    billing_address: {
      id: "test-billing-address",
      first_name: "lebron",
    },
    shipping_address: {
      id: "test-shipping-address",
      first_name: "lebron",
      country_code: "us",
    },
    region_id: "test-region",
    currency_code: "usd",
    tax_rate: 0,
    discounts: [],
    payments: [
      {
        id: "test-payment-for-claim-order",
        amount: 10000,
        currency_code: "usd",
        amount_refunded: 0,
        provider_id: "test-pay",
        data: {},
      },
    ],
    items: [],
    ...data,
  })

  await manager.save(orderWithClaim)

  const li = manager.create(LineItem, {
    id: "test-item-co-2",
    fulfilled_quantity: 1,
    title: "Line Item",
    description: "Line Item Desc",
    thumbnail: "https://test.js/1234",
    unit_price: 8000,
    quantity: 1,
    variant_id: "test-variant",
    order_id: orderWithClaim.id,
  })

  await manager.save(li)

  const li2 = manager.create(LineItem, {
    id: "test-item-co-3",
    fulfilled_quantity: 4,
    title: "Line Item",
    description: "Line Item Desc",
    thumbnail: "https://test.js/1234",
    unit_price: 8000,
    quantity: 4,
    variant_id: "test-variant",
    order_id: orderWithClaim.id,
  })

  await manager.save(li2)

  const claimWithFulfillment = manager.create(ClaimOrder, {
    id: "claim-w-f",
    type: "replace",
    payment_status: "na",
    fulfillment_status: "not_fulfilled",
    order_id: "order-with-claim",
    ...data,
  })

  const ful1 = manager.create(Fulfillment, {
    id: "fulfillment-co-1",
    data: {},
    provider_id: "test-ful",
  })

  const ful2 = manager.create(Fulfillment, {
    id: "fulfillment-co-2",
    data: {},
    provider_id: "test-ful",
  })

  claimWithFulfillment.fulfillments = [ful1, ful2]

  await manager.save(claimWithFulfillment)

  const claimWithReturn = manager.create(ClaimOrder, {
    id: "claim-w-r",
    type: "replace",
    payment_status: "na",
    fulfillment_status: "not_fulfilled",
    order_id: "order-with-claim",
    ...data,
  })

  await manager.save(claimWithReturn)

  await manager.insert(Return, {
    id: "return-id-2",
    claim_order_id: "claim-w-r",
    status: "requested",
    refund_amount: 0,
    data: {},
  })
}
