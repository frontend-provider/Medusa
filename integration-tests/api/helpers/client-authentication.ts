const AUTH_COOKIE = {}
export async function getClientAuthenticationCookie(
  api,
  email = null,
  password = null
) {
  const user = {
    email: email ?? "test@medusajs.com",
    password: password ?? "test",
  }

  if (AUTH_COOKIE[user.email]) {
    return AUTH_COOKIE[user.email]
  }

  const authResponse = await api.post("/store/auth", user)
  AUTH_COOKIE[user.email] = authResponse.headers["set-cookie"][0].split(";")

  return AUTH_COOKIE[user.email]
}
