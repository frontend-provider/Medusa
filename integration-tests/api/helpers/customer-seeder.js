const { Customer, Address } = require("@medusajs/medusa")

module.exports = async (connection, data = {}) => {
  const manager = connection.manager

  await manager.insert(Customer, {
    id: "test-customer-1",
    email: "test1@email.com",
  })

  await manager.insert(Customer, {
    id: "test-customer-2",
    email: "test2@email.com",
  })

  await manager.insert(Customer, {
    id: "test-customer-3",
    email: "test3@email.com",
  })

  await manager.insert(Customer, {
    id: "test-customer-has_account",
    email: "test4@email.com",
    has_account: true,
  })

  await manager.insert(Address, {
    id: "test-address",
    first_name: "Lebron",
    last_name: "James",
    customer_id: "test-customer-1",
  })
}
