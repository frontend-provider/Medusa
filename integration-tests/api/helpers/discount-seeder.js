const {
  ShippingProfile,
  Region,
  Discount,
  DiscountRule,
} = require("@medusajs/medusa")
module.exports = async (dataSource, data = {}) => {
  const manager = dataSource.manager

  await manager.insert(Region, {
    id: "test-region",
    name: "Test Region",
    currency_code: "usd",
    tax_rate: 0,
    payment_providers: [
      {
        id: "test-pay",
        is_installed: true,
      },
    ],
  })

  await manager.insert(Region, {
    id: "test-region-2",
    name: "Test Region 2",
    currency_code: "eur",
    tax_rate: 0,
    payment_providers: [
      {
        id: "test-pay",
        is_installed: true,
      },
    ],
  })
}
