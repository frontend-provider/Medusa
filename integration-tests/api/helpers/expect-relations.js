const { expect } = require("@jest/globals")

export function expectRelations(expected, actual){
    expected.forEach(a => expectRelations_(a, actual))
}

function expectRelations_(expected, actual){
    const items = expected.split(".");
    let data_ = actual;

    for(const i in items){
        data_ = data_[items[i]];
        
        if(data_ instanceof Array && data_.length > 0) {
            data_ = data_[0];
        } 
    
        expect(data_).toBeDefined();
    }
}
