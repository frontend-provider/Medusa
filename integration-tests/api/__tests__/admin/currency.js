const path = require("path")
const setupServer = require("../../../helpers/setup-server")
const startServerWithEnvironment =
  require("../../../helpers/start-server-with-environment").default
const { useApi } = require("../../../helpers/use-api")
const { useDb, initDb } = require("../../../helpers/use-db")
const adminSeeder = require("../../helpers/admin-seeder")

const adminReqConfig = {
  headers: {
    Authorization: "Bearer test_token",
  },
}

jest.setTimeout(30000)
describe("/admin/currencies", () => {
  let medusaProcess
  let dbConnection

  beforeAll(async () => {
    const cwd = path.resolve(path.join(__dirname, "..", ".."))
    dbConnection = await initDb({ cwd })
    medusaProcess = await setupServer({ cwd })
  })

  afterAll(async () => {
    const db = useDb()
    await db.shutdown()

    medusaProcess.kill()
  })

  describe("GET /admin/currencies", function () {
    beforeEach(async () => {
      try {
        await adminSeeder(dbConnection)
      } catch (e) {
        console.error(e)
      }
    })

    afterEach(async () => {
      const db = useDb()
      await db.teardown()
    })

    it("should retrieve the currencies", async () => {
      const api = useApi()
      const response = await api.get(
        `/admin/currencies?order=code`,
        adminReqConfig
      )

      expect(response.data).toMatchSnapshot()
    })
  })
})
describe("[MEDUSA_FF_TAX_INCLUSIVE_PRICING] /admin/currencies", () => {
  let medusaProcess
  let dbConnection

  beforeAll(async () => {
    const cwd = path.resolve(path.join(__dirname, "..", ".."))
    const [process, connection] = await startServerWithEnvironment({
      cwd,
      env: { MEDUSA_FF_TAX_INCLUSIVE_PRICING: true },
    })
    dbConnection = connection
    medusaProcess = process
  })

  afterAll(async () => {
    const db = useDb()
    await db.shutdown()

    medusaProcess.kill()
  })

  describe("GET /admin/currencies", function () {
    beforeEach(async () => {
      try {
        await adminSeeder(dbConnection)
      } catch (e) {
        console.error(e)
      }
    })

    afterEach(async () => {
      const db = useDb()
      await db.teardown()
    })

    it("should retrieve the currencies", async () => {
      const api = useApi()
      const response = await api.get(
        `/admin/currencies?order=code`,
        adminReqConfig
      )

      expect(response.data).toMatchSnapshot()
    })
  })

  describe("POST /admin/currencies/:code", function () {
    beforeEach(async () => {
      try {
        await adminSeeder(dbConnection)
      } catch (e) {
        console.error(e)
      }
    })

    afterEach(async () => {
      const db = useDb()
      await db.teardown()
    })

    it("should update currency includes_tax", async () => {
      const api = useApi()
      const response = await api.post(
        `/admin/currencies/aed`,
        {
          includes_tax: true,
        },
        adminReqConfig
      )

      expect(response.data).toMatchSnapshot()
    })
  })
})
