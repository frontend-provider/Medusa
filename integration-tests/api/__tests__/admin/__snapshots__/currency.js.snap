// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/admin/currencies GET /admin/currencies should retrieve the currencies 1`] = `
Object {
  "count": 120,
  "currencies": Array [
    Object {
      "code": "aed",
      "name": "United Arab Emirates Dirham",
      "symbol": "AED",
      "symbol_native": "د.إ.‏",
    },
    Object {
      "code": "afn",
      "name": "Afghan Afghani",
      "symbol": "Af",
      "symbol_native": "؋",
    },
    Object {
      "code": "all",
      "name": "Albanian Lek",
      "symbol": "ALL",
      "symbol_native": "Lek",
    },
    Object {
      "code": "amd",
      "name": "Armenian Dram",
      "symbol": "AMD",
      "symbol_native": "դր.",
    },
    Object {
      "code": "ars",
      "name": "Argentine Peso",
      "symbol": "AR$",
      "symbol_native": "$",
    },
    Object {
      "code": "aud",
      "name": "Australian Dollar",
      "symbol": "AU$",
      "symbol_native": "$",
    },
    Object {
      "code": "azn",
      "name": "Azerbaijani Manat",
      "symbol": "man.",
      "symbol_native": "ман.",
    },
    Object {
      "code": "bam",
      "name": "Bosnia-Herzegovina Convertible Mark",
      "symbol": "KM",
      "symbol_native": "KM",
    },
    Object {
      "code": "bdt",
      "name": "Bangladeshi Taka",
      "symbol": "Tk",
      "symbol_native": "৳",
    },
    Object {
      "code": "bgn",
      "name": "Bulgarian Lev",
      "symbol": "BGN",
      "symbol_native": "лв.",
    },
    Object {
      "code": "bhd",
      "name": "Bahraini Dinar",
      "symbol": "BD",
      "symbol_native": "د.ب.‏",
    },
    Object {
      "code": "bif",
      "name": "Burundian Franc",
      "symbol": "FBu",
      "symbol_native": "FBu",
    },
    Object {
      "code": "bnd",
      "name": "Brunei Dollar",
      "symbol": "BN$",
      "symbol_native": "$",
    },
    Object {
      "code": "bob",
      "name": "Bolivian Boliviano",
      "symbol": "Bs",
      "symbol_native": "Bs",
    },
    Object {
      "code": "brl",
      "name": "Brazilian Real",
      "symbol": "R$",
      "symbol_native": "R$",
    },
    Object {
      "code": "bwp",
      "name": "Botswanan Pula",
      "symbol": "BWP",
      "symbol_native": "P",
    },
    Object {
      "code": "byn",
      "name": "Belarusian Ruble",
      "symbol": "Br",
      "symbol_native": "руб.",
    },
    Object {
      "code": "bzd",
      "name": "Belize Dollar",
      "symbol": "BZ$",
      "symbol_native": "$",
    },
    Object {
      "code": "cad",
      "name": "Canadian Dollar",
      "symbol": "CA$",
      "symbol_native": "$",
    },
    Object {
      "code": "cdf",
      "name": "Congolese Franc",
      "symbol": "CDF",
      "symbol_native": "FrCD",
    },
  ],
  "limit": 20,
  "offset": 0,
}
`;

exports[`[MEDUSA_FF_TAX_INCLUSIVE_PRICING] /admin/currencies GET /admin/currencies should retrieve the currencies 1`] = `
Object {
  "count": 120,
  "currencies": Array [
    Object {
      "code": "aed",
      "includes_tax": false,
      "name": "United Arab Emirates Dirham",
      "symbol": "AED",
      "symbol_native": "د.إ.‏",
    },
    Object {
      "code": "afn",
      "includes_tax": false,
      "name": "Afghan Afghani",
      "symbol": "Af",
      "symbol_native": "؋",
    },
    Object {
      "code": "all",
      "includes_tax": false,
      "name": "Albanian Lek",
      "symbol": "ALL",
      "symbol_native": "Lek",
    },
    Object {
      "code": "amd",
      "includes_tax": false,
      "name": "Armenian Dram",
      "symbol": "AMD",
      "symbol_native": "դր.",
    },
    Object {
      "code": "ars",
      "includes_tax": false,
      "name": "Argentine Peso",
      "symbol": "AR$",
      "symbol_native": "$",
    },
    Object {
      "code": "aud",
      "includes_tax": false,
      "name": "Australian Dollar",
      "symbol": "AU$",
      "symbol_native": "$",
    },
    Object {
      "code": "azn",
      "includes_tax": false,
      "name": "Azerbaijani Manat",
      "symbol": "man.",
      "symbol_native": "ман.",
    },
    Object {
      "code": "bam",
      "includes_tax": false,
      "name": "Bosnia-Herzegovina Convertible Mark",
      "symbol": "KM",
      "symbol_native": "KM",
    },
    Object {
      "code": "bdt",
      "includes_tax": false,
      "name": "Bangladeshi Taka",
      "symbol": "Tk",
      "symbol_native": "৳",
    },
    Object {
      "code": "bgn",
      "includes_tax": false,
      "name": "Bulgarian Lev",
      "symbol": "BGN",
      "symbol_native": "лв.",
    },
    Object {
      "code": "bhd",
      "includes_tax": false,
      "name": "Bahraini Dinar",
      "symbol": "BD",
      "symbol_native": "د.ب.‏",
    },
    Object {
      "code": "bif",
      "includes_tax": false,
      "name": "Burundian Franc",
      "symbol": "FBu",
      "symbol_native": "FBu",
    },
    Object {
      "code": "bnd",
      "includes_tax": false,
      "name": "Brunei Dollar",
      "symbol": "BN$",
      "symbol_native": "$",
    },
    Object {
      "code": "bob",
      "includes_tax": false,
      "name": "Bolivian Boliviano",
      "symbol": "Bs",
      "symbol_native": "Bs",
    },
    Object {
      "code": "brl",
      "includes_tax": false,
      "name": "Brazilian Real",
      "symbol": "R$",
      "symbol_native": "R$",
    },
    Object {
      "code": "bwp",
      "includes_tax": false,
      "name": "Botswanan Pula",
      "symbol": "BWP",
      "symbol_native": "P",
    },
    Object {
      "code": "byn",
      "includes_tax": false,
      "name": "Belarusian Ruble",
      "symbol": "Br",
      "symbol_native": "руб.",
    },
    Object {
      "code": "bzd",
      "includes_tax": false,
      "name": "Belize Dollar",
      "symbol": "BZ$",
      "symbol_native": "$",
    },
    Object {
      "code": "cad",
      "includes_tax": false,
      "name": "Canadian Dollar",
      "symbol": "CA$",
      "symbol_native": "$",
    },
    Object {
      "code": "cdf",
      "includes_tax": false,
      "name": "Congolese Franc",
      "symbol": "CDF",
      "symbol_native": "FrCD",
    },
  ],
  "limit": 20,
  "offset": 0,
}
`;

exports[`[MEDUSA_FF_TAX_INCLUSIVE_PRICING] /admin/currencies POST /admin/currencies/:code should update currency includes_tax 1`] = `
Object {
  "currency": Object {
    "code": "aed",
    "includes_tax": true,
    "name": "United Arab Emirates Dirham",
    "symbol": "AED",
    "symbol_native": "د.إ.‏",
  },
}
`;
